pandas/core/arrays/datetimes.py

```diff
@@ -1568,6 +1633,8 @@ def sequence_to_dt64ns(data, dtype=None, copy=False,
         copy = False
     elif isinstance(data, ABCSeries):
         data = data._values
+    elif isinstance(data, ABCPandasArray):
+        data = data._ndarray
```

Comment:
This may be unnecessary... Do you recall why it was added @jbrockmendel? There was a brief time when `Series[datetime64[ns]].array` was a PandasArray and this may have mattered. Otherwise I assume we should eventually call np.asarray(data) to convert the sequence to an ndarray?

Reply:
In one of the rebasing PR^2 there were tests failing because the end of this functions asserts it is returning an ndarray and instead it was returning a PandasArray.  I didn't have any particular reason to use _ndarray instead of another method

Reply:
I'l cahnage this to `to_numpy()`.

Looks like we have some trouble with non-ndarrays: https://github.com/pandas-dev/pandas/issues/24539